### DISCUSSION TITLE
Search content, complete with file name

### DISCUSSION THREAD

**Discussion by ghost** [ibhagwan/fzf-lua#1151] [Q&A] [answered]
I want to search files for content or symbols and turn the command into a complete command, but with the file name for the selected result instead of the full result. Is the best way to do that:

```lua
require("fzf-lua").fzf_exec({"foo", "bar"}, {
   complete = function(selected, opts, line, col)
    ----------------------------
    -- Parse selection  to strip it of everything but file name
    ----------------------------
    -- then:
    local newline = line:sub(1, col) .. selected[1]
    return newline, #newline - 1
  end
})
```

edit: I've marked the answer that works for me (on Windows), but for Linux the solution in these comments might be cleaner:

https://github.com/ibhagwan/fzf-lua/discussions/1151#discussioncomment-9246314
https://github.com/ibhagwan/fzf-lua/discussions/1151#discussioncomment-9246376

**Comment by ibhagwan** [comment, marked as answer]
- Not sure how you’re extracting the file name but you can use `path.file_to_entry`
- The code is completing the line with the selection and setting the cursor to after the completion, this seems like the desirable behavior to me, if that’s what you’re after I don’t think there’s a better way.

**Comment by ghost** [comment]
I found this, which is probably a better description of what I want: https://superuser.com/questions/1632357/how-do-i-configure-fzf-to-return-only-a-specific-field-of-the-selected-line. 

```shell
ps -fe | fzf --bind 'enter:become(echo {2})'
```
```shell
ps -fe | fzf -m --bind 'enter:become(echo {+2})'
```

I will see if I can get working with fzf-lua's options.

**Comment by ibhagwan** [comment]
> I will see if I can get working with fzf-lua's options.

You can either pass these directly using `fzf_cli_args`, use `keymap.fzf['enter']`

**Comment by ghost** [comment]
```shell
❯ ls | fzf --bind 'enter:become(echo {2})'
become action is not supported on Windows
```

Welp. Back to parsing the result in the complete function then.

edit: btw, thank you so much for getting fzflua working on Windows. Been wishing for that for a long time.

**Reply by ibhagwan** [reply]
Fzf doesn’t like windows lol

**Reply by ibhagwan** [reply]
FYI, fzf 0.51 was just released with:
```
become action is now supported on Windows
```

**Reply by ghost** [reply]
Great timing for me 😄
